source/ID/concordium-did.rst

```diff
@@ -0,0 +1,676 @@
+            "controller": "did:ccd:NET:cred:CRED",
+            "publicKeyMultibase": "fXX"
+          },
+          "...",
+          {
+            "id": "did:ccd:NET:cred:CRED#key-N",
+            "type": "Ed25519VerificationKey2020",
+            "controller": "did:ccd:NET:cred:CRED",
+            "publicKeyMultibase": "fYY"
+          }
+        ]
+      }
+    ],
+    "authentication": [
+      "#credential-vm"
+    ]
+  }
+
+
+Smart Contract Instance DID
+---------------------------
+
+The goal of the Smart Contract Instance DID is to provide meta-data about the contract instance.
+At the moment, it contains an account address of the initialization transaction sender, and the list of the contract's entrypoints.
+
+The Smart Contract Instance DID Document MUST contain the following data:
+
+- ``id`` - the DID of the smart contract instance.
+- ``creator`` - a DID of an account that initialized the contract instance represented as a JSON object containing fields ``id`` and ``account``.
+- ``entrypoints`` - a list on the contract's entrypoints. Each entrypoint is an object containing fields ``id`` and ``name``.
+
+The document MAY include any other public data of a smart contract instance.
+
+.. code-block:: json
+
+  {
+    "id": "did:ccd:sci:IND:SUBIND",
+    "owner": {
+      "id": "did:ccd:sci:IND:SUBIND#creator",
+      "account": "did:ccd:NET:acc:ADDR"
+    }
+    "entrypoints": [
+      { "id": "did:ccd:sci:IND:SUBIND#entrypoint-issuerKeys",
+        "name": "issuerKeys"
+      },
+      { "id": "did:ccd:sci:IND:SUBIND#entrypoint-revocationKey",
+        "name": "revocationKey"
+      }
+    ]
+  }
+
+Where ``IND`` and ``SUBIND`` are the contract index and subindex.
+``NET`` and ``ADDR`` correspond to the network and to the owner's account address.
+
+
+.. _concordium-did-pkc:
+
+Public Key Cryptography DID
+---------------------------
+
+The goal of the Public Key Cryptography DID is to represent a public key and the corresponding signature verification method.
+
+The Public Key Cryptography DID Document MUST contain the following data:
+
+- ``id`` - the DID of the public key.
+- ``verificationMethod`` - specifies a `DID verification method <did-vefication-method_>`_ for verifying a signature corresponding to the public key.
+- ``authentication`` - authentication method for the key.
+
+.. code-block:: json
+
+  {
+    "id": "did:ccd:pkc:XX",
+    "verificationMethod": [
+      {
+        "id": "did:ccd:pkc:XX#key-0",
+        "type": "Ed25519VerificationKey2020",
+        "controller": "did:ccd:NET:pkc:PK",
+        "publicKeyMultibase": "fXX"
+      }
+    ],
+    "authentication": [
+      {
+        "did:ccd:pkc:XX#key-0"
+      }
+    ]
+  }
+
+Identity Provider DID
+---------------------
+
+The goal of the Identity Provider DID is identify a Concordium identity provider.
+An identity provider is an organization, approved by Concordium, that performs off-chain identification of users.
```

Comment:
```suggestion
An identity provider (IDP) is an organization, approved by Concordium, that performs off-chain identification of users.
```

Reply:
@abizjak suggested [here](https://github.com/Concordium/concordium-update-proposals/pull/39/files/35062a110099c3074be5c0151507facf54bc6c83#r1222690694) not use the acronym.
I guess we could use "IDP" if it's recognizable, that is, used in other parts of our documentation.

Reply:
It is written out everywhere in the documentation. But the general rule is that as long as you write it out the first time and follow it in parentheses with the acronym, then it's fine to use the acronym in the rest of the document. But I will leave the final decision to you and Ales.

Reply:
That was why I put in IDP next to this. Because IDP is used elsewhere in the same doc.

Reply:
> That was why I put in IDP next to this. Because IDP is used elsewhere in the same doc.

@dg-concordium you're right, I thought I expanded the acronym in all places in the document, but it's used in several places, my bad. So, either I spell it out in full everywhere or introduce an acronym.

Reply:
@dg-concordium I'll add the acronym in parentheses, as you suggested. It's used many times, so it makes sense to have it.